evolver/tests/test_base.py

```diff
@@ -280,6 +286,29 @@ def test_from_file(self, tmp_path, mock_descriptor):
         assert obj.a == 11
         assert obj.b == 22
 
+    @pytest.mark.parametrize("shallow", (True, False))
+    def test_nested(self, monkeypatch, shallow):
+        if not shallow:
+            monkeypatch.setattr(evolver.base._BaseConfig, "shallow_model_dump", lambda x: x.model_dump())
+
```

Comment:
why is this necessary, since we don't really have a formal option of shallow-or-deep?

Reply:
Yeah, it's a bit odd I agree, I wanted to add a test to show when this fails to ensure that the test does actually test the success since it's a bit of a narrow gap to test.

``model_dump()`` is "deep" so I monkey patch it back to a deep conversion to ensure that it fails. If it doesn't fail, the test isn't adequate.